File: 2167-minimum-time-to-remove-all-cars-containing-illegal-goods/2167-minimum-time-to-remove-all-cars-containing-illegal-goods.py
class Solution:
    def minimumTime(self, s: str) -> int:
        c = 0
        x = 0
        for i in s:
            if i == '1':
                c += 1
            else:
                c -= 1
            if c < x:
                x = c
            if c > 0:
                c = 0
        return len(s) + x

File: 2167-minimum-time-to-remove-all-cars-containing-illegal-goods/2167-minimum-time-to-remove-all-cars-containing-illegal-goods.cpp
class Solution {
public:
    int minimumTime(string s) {
        int c = 0;
        int x = 0;
        for (char i: s) {
            if (i == '1') {
                c++;
            } else {
                c--;
            }
            if (c < x) {
                x = c;
            }
            if (c > 0) {
                c = 0;
            }
        }
        return s.length() + x;
    }
};